bundle/config/mutator/override_compute.go

```diff
@@ -38,24 +38,33 @@ func overrideJobCompute(j *resources.Job, compute string) {
 }
 
 func (m *overrideCompute) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
-	if b.Config.Bundle.Mode != config.Development {
+	var diags diag.Diagnostics
+
+	if b.Config.Bundle.Mode == config.Production {
 		if b.Config.Bundle.ClusterId != "" {
-			return diag.Errorf("cannot override compute for an target that does not use 'mode: development'")
+			// Overriding compute via a command-line flag for production works, but is not recommended.
+			diags = diags.Extend(diag.Warningf("overriding compute for a target that uses 'mode: production' is not recommended"))
+		}
+		if env.Get(ctx, "DATABRICKS_CLUSTER_ID") != "" {
+			// The DATABRICKS_CLUSTER_ID may be set by accident when doing a production deploy.
+			// Overriding the cluster in production is almost always a mistake since customers
+			// want consistency in production and not compute that is different each deploy.
+			// For this reason we log a warning and ignore the environment variable.
+			return diag.Warningf("the DATABRICKS_CLUSTER_ID variable is set but is ignored since the current target uses 'mode: production'")
```

Comment:
This introduces a bifurcation since `DATABRICKS_CLUSTER_ID` now does not have a 1:1 correspondence to `b.Config.Bundle.ClusterId`. 

Can we instead support the environment variable but extend the warning to let the users know they are relying on the environment variable which is not recommended for mode production and instead they should use `bundle.cluster_id`?

Reply:
We previously ignored DATABRICKS_CLUSTER_ID for all non-development targets. The new version just makes sure that we're loud about what happens.

Reply:
Yeah, but we also did not support `--cluster-id` or `b.Config.Bundle.ClusterId` for non-development targets as well. 

If `DATABRICKS_CLUSTER_ID` is just the environment variable equivalent for the above fields then we should consider not inducing this bifurcation in functionality. 

@pietern what do you think?

Reply:
Using an environment variable to toggle behavior was a mistake in retrospect and we shouldn't expand its use beyond its current use in development mode, IMO.

The net-new behavior for this change can be limited to just the `--cluster-id` flag.

Reply:
> we'd allow --cluster-id for all targets. And warn for mode: production.

Indeed.

> DATABRICKS_CLUSTER_ID is indeed rather prone to mistakes. How about we always show an info message when it is used? (or a warning?)

This is limited to `mode: development` today and I don't want to make it more generally usable. I propose we continue to use the env var _only_ in development mode _and_ we include an info message when it is picked up from the env var. That makes it backward-compatible without expanding its use. For example: users who have already set it in their env for unrelated reasons, maybe Databricks Connect, and are not using `mode: development`; if we start to use it now, they will observe a behavior change.

Reply:
👍 updated accordingly. The new version essentially just extends `--cluster-id` and adds more warnings/info messages and tests.